inline_combination without return value?

Is this code valid as an inline combinational code?
(just deleted `O1 = reg.O` lines from tests/test_syntax/test_inline_comb.py)

    @m.inline_combinational()
    def logic():
        if io.invert:
            reg.I @= ~reg.O
        else:
            reg.I @= reg.O

If so, how about this? using `m.wire` instead of `@=`.

    @m.inline_combinational()
    def logic():
        if io.invert:
            m.wire(~reg.O, reg.I)
        else:
            m.wire(reg.O, reg.I)

both are same code but the latter causes error, because the logic function doesn't return value.

    >       retval = ast.Tuple(elts, ast.Load()) if len(elts) > 1 else elts[0]
    E       IndexError: list index out of range

--- a/magma/syntax/inline_combinational.py
+++ b/magma/syntax/inline_combinational.py
@@ -10,16 +10,18 @@ from ast_tools.passes import (apply_ast_passes, Pass, PASS_ARGS_T, ssa,
 from ast_tools.common import gen_free_prefix, gen_free_name
 from ast_tools.immutable_ast import immutable, mutable
 
+import magma as magma_module
 from magma.syntax.combinational2 import combinational2
 from magma.bit import Bit
 from magma.wire import wire
 
 
 class _ToCombinationalRewriter(ast.NodeTransformer):
-    def __init__(self, free_prefix, wire_map, assign_map):
+    def __init__(self, free_prefix, wire_map, assign_map, env):
         self.prefix = free_prefix
         self.wire_map = wire_map
         self.assign_map = assign_map
+        self.env = env
         self.name_count = 0
 
     def _gen_new_name(self):
@@ -27,14 +29,17 @@ class _ToCombinationalRewriter(ast.NodeTransformer):
         self.name_count += 1
         return new_name
 
-    def visit_AugAssign(self, node):
-        if not isinstance(node.op, ast.MatMult):
-            return node
-        key = immutable(node.target)
+    def _gen_assign(self, target, value):
+        key = immutable(target)
         if key not in self.wire_map:
             self.wire_map[key] = self._gen_new_name()
         target = ast.Name(self.wire_map[key], ast.Store())
-        return ast.Assign([target], node.value)
+        return ast.Assign([target], value)
+
+    def visit_AugAssign(self, node):
+        if not isinstance(node.op, ast.MatMult):
+            return node
+        return self._gen_assign(node.target, node.value)
 
     def visit_Assign(self, node):
         for i, target in enumerate(node.targets):
@@ -42,6 +47,15 @@ class _ToCombinationalRewriter(ast.NodeTransformer):
             if key not in self.assign_map:
                 self.assign_map[key] = self._gen_new_name()
             node.targets[i] = ast.Name(self.assign_map[key], ast.Store())
+        return node
+
+    def visit_Call(self, node):
+        if (
+            isinstance(node.func, ast.Attribute) and node.func.attr == "wire"
+            and eval(node.func.value.id, {}, self.env) is magma_module
+        ):
+            # m.wire(value, target)
+            return self._gen_assign(node.args[1], node.args[0])
         return node
 
 
@@ -66,7 +80,7 @@ class _RewriteToCombinational(Pass):
         prefix = gen_free_prefix(tree, env)
 
         tree = _ToCombinationalRewriter(prefix, self.wire_map,
-                                        self.assign_map).visit(tree)
+                                        self.assign_map, env).visit(tree)
 
         # Return targets for wiring and assignments
         return_values = (list(self.wire_map.values()) +

--- a/tests/test_syntax/gold/test_inline_comb_wire.v
+++ b/tests/test_syntax/gold/test_inline_comb_wire.v
@@ -1,0 +1,73 @@
+module coreir_reg #(
+    parameter width = 1,
+    parameter clk_posedge = 1,
+    parameter init = 1
+) (
+    input clk,
+    input [width-1:0] in,
+    output [width-1:0] out
+);
+  reg [width-1:0] outReg=init;
+  wire real_clk;
+  assign real_clk = clk_posedge ? clk : ~clk;
+  always @(posedge real_clk) begin
+    outReg <= in;
+  end
+  assign out = outReg;
+endmodule
+
+module Register (
+    input I,
+    output O,
+    input CLK
+);
+wire [0:0] reg_P_inst0_out;
+coreir_reg #(
+    .clk_posedge(1'b1),
+    .init(1'h0),
+    .width(1)
+) reg_P_inst0 (
+    .clk(CLK),
+    .in(I),
+    .out(reg_P_inst0_out)
+);
+assign O = reg_P_inst0_out[0];
+endmodule
+
+module Mux2xOutBit (
+    input I0,
+    input I1,
+    input S,
+    output O
+);
+reg [0:0] coreir_commonlib_mux2x1_inst0_out;
+always @(*) begin
+if (S == 0) begin
+    coreir_commonlib_mux2x1_inst0_out = I0;
+end else begin
+    coreir_commonlib_mux2x1_inst0_out = I1;
+end
+end
+
+assign O = coreir_commonlib_mux2x1_inst0_out[0];
+endmodule
+
+module Main (
+    input invert,
+    output O,
+    input CLK
+);
+wire Mux2xOutBit_inst0_O;
+Mux2xOutBit Mux2xOutBit_inst0 (
+    .I0(O),
+    .I1(~ O),
+    .S(invert),
+    .O(Mux2xOutBit_inst0_O)
+);
+Register Register_inst0 (
+    .I(Mux2xOutBit_inst0_O),
+    .O(O),
+    .CLK(CLK)
+);
+endmodule
+

--- a/tests/test_syntax/test_inline_comb.py
+++ b/tests/test_syntax/test_inline_comb.py
@@ -23,3 +23,25 @@ def test_inline_comb_basic():
     m.compile("build/test_inline_comb_basic", Main, inline=True)
     assert check_files_equal(__file__, f"build/test_inline_comb_basic.v",
                              f"gold/test_inline_comb_basic.v")
+
+
+def test_inline_comb_wire():
+    import magma as magma_test
+
+    class Main(magma_test.Circuit):
+        io = magma_test.IO(invert=magma_test.In(magma_test.Bit), O=magma_test.Out(magma_test.Bit))
+        io += magma_test.ClockIO()
+        reg = magma_test.Register(magma_test.Bit)()
+
+        @magma_test.inline_combinational(debug=True, file_name="test_inline_comb_wire.py")
+        def logic():
+            if io.invert:
+                magma_test.wire(~reg.O, reg.I)
+            else:
+                magma_test.wire(reg.O, reg.I)
+
+        io.O @= reg.O
+
+    magma_test.compile("build/test_inline_comb_wire", Main, inline=True)
+    assert check_files_equal(__file__, f"build/test_inline_comb_wire.v",
+                             f"gold/test_inline_comb_wire.v")

